src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -355,7 +355,7 @@ private static void WritePathSegment(CodeWriter writer, CodeWriterDeclaration ur
         {
             methodName ??= segment.IsRaw ? "AppendRaw" : "AppendPath";
             writer.Append($"{uri}.{methodName}(");
-            WriteConstantOrParameter(writer, value, enumAsString: !segment.IsRaw || value.Type is { IsFrameworkType: false, Implementation: EnumType { IsExtensible: true } });
```

Comment:
in `cadl-ranch` `multi-client` case, client type is now a fixed enum. So when composing request, we should generate codes like `appendRaw(_client.ToSerial())`, not `appendRaw(_client)`